include/seqan3/alphabet/concept.hpp

```diff
@@ -188,24 +237,28 @@ namespace seqan3
  * This is a customisation point (see \ref about_customisation). To specify the behaviour for your own alphabet type,
  * simply provide one of the three functions specified above.
  */
-inline constexpr auto assign_rank_to = detail::adl::only::assign_rank_to_fn{};
+inline constexpr auto assign_rank_to = detail::adl_only::assign_rank_to_fn{};
 //!\}
 } // namespace seqan3
 
 // ============================================================================
 // to_char()
 // ============================================================================
 
-namespace seqan3::detail::adl::only
+namespace seqan3::detail::adl_only
 {
 
+//!\brief Poison-pill overload to prevent non-ADL forms of unqualified lookup.
+template <typename ...args_t>
+void to_char(args_t ...) = delete;
```

Comment:
```suggestion
void to_char(...) = delete;
```

And remove template argument list as well as update all other places